src/sentry/tasks/merge.py

```diff
@@ -257,11 +257,16 @@ def merge_objects(models, group, new_group, limit=1000,
                             )
                     elif model == GroupTagValue:
                         with transaction.atomic(using=router.db_for_write(model)):
-                            model.objects.filter(
+                            new_obj = model.objects.get(
```

Comment:
Barring any race conditions, moving this to `get` should be safe since we've already established the other object exists (otherwise we wouldn't be in this block) and the query conditions are all part of the same unique constraint.